tests/test_ReplayBuffer.py

```diff
@@ -50,6 +50,32 @@ def test_simple_buffer(self):
         unique.sort()
         assert np.all(unique == np.array([2, 3, 4, 5, 6]))
 
+    def test_without_replacement(self):
+        rng = np.random.default_rng(0)
+        buffer = ReplayBuffer(5, 1, rng)
+
+        # on creation, the buffer should have no size
+        assert buffer.size() == 0
+
+        # should be able to simply add and sample a single data point
+        d = fake_timestep(a=-1, r=None)
+        buffer.add_step(d)
+        assert buffer.size() == 0
+
+        for a in range(5):
+            d = fake_timestep(a=a)
+            buffer.add_step(d)
+
+        assert buffer.size() == 5
+
+        for _ in range(25):
+            batch = buffer.sample_without_replacement(3)
+            assert len(set(batch.a)) == 3
```

Comment:
Because each action in this test is unique (-1, 0, 1, 2, ...) then we can check for without-replacement sampling by checking if we get 3 unique actions. Conversion to a set guarantees no duplicates.

While this is technically a fuzz test, the probability of getting 3 unique actions using with-replacement 25 times in a row is astronomically small.